### PR TITLE
Added get app by name route

### PR DESCRIPTION
**Pull request by AnirudhPokala**
To get the a particular app by its name.
Screenshot 2021-11-30 at 5 26 12 PM

### CODE REVIEW

File: pkg/api/api.go [hunk 1]
```diff
@@ -80,3 +81,23 @@ func createApp(w http.ResponseWriter, r *http.Request) {
...
+	vars := mux.Vars(r)
+	spaceName := vars["space"]
+	appName := vars["name"]
+
+	appList, err := knative.GetAppByName(util.Kubeconfig, spaceName, appName)
+
+	if err != nil {
+		log.Error(err, "while listing app")
+		w.WriteHeader(http.StatusInternalServerError)
+		return
```
Comment (nikhilbandi) [inline pkg/api/api.go:95]: Shouldn't we return error to calling function?

File: pkg/api/api.go [hunk 2]
```diff
@@ -80,3 +81,23 @@ func createApp(w http.ResponseWriter, r *http.Request) {
 	}
 	w.WriteHeader(http.StatusOK)
 }
+
+func getAppByName(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	spaceName := vars["space"]
```
Comment (nikhilbandi) [inline pkg/api/api.go:87]: Variable name 'space' looks more relevant.

File: pkg/api/api.go (hunk 1 above)
Comment (AnirudhPokala) [reply pkg/api/api.go:95]: We are logging it, and also writing error with Response writer.

Comment (nikhilbandi) [reply pkg/api/api.go:95]: OK

**Review by nikhilbandi — APPROVED** [review]
Approving.

**Pull request merged**